### PR TITLE
don't wait for sample image JPEG to complete

### LINKED ISSUE #399: blackfly_optical should not block RE execution
Do this measurement in a background thread.  No need for data acquisition to wait.

### PR DESCRIPTION
**Pull request by prjemian**
fixes #399

### CODE REVIEW

File: profile_bluesky/startup/instrument/devices/blackfly.py [hunk 1]
```diff
@@ -76,9 +78,14 @@ def should_save_jpeg(self):
         return _flag_save_sample_image_jpeg_.get() in (1, "Yes")
 
     def take_image(self):
-        yield from bps.stage(self)
-        yield from bps.trigger(self, wait=True)
```
Comment (prjemian) [inline profile_bluesky/startup/instrument/devices/blackfly.py:80]: Could we have done the same thing by removing the `wait=True` keyword argument?  I don't think so but we could try that.

Comment (prjemian) [reply profile_bluesky/startup/instrument/devices/blackfly.py:80]: setting `wait=False` does not execute properly but the branch seems to do the right job

**Comment by prjemian** [comment]
This is not ready yet.  Cannot unstage until acquisition is complete.

File: profile_bluesky/startup/instrument/devices/blackfly.py [hunk 2]
```diff
@@ -76,9 +78,14 @@ def should_save_jpeg(self):
...
-        yield from bps.unstage(self)
+        @apstools.utils.run_in_thread
+        def _snapshot():
+            self.stage()
+            self.trigger()
+            self.unstage()
```
Comment (prjemian) [inline profile_bluesky/startup/instrument/devices/blackfly.py:85]: need to wait for acquire to finish

**Comment by prjemian** [comment]
This needs to operate like the `pi_c867` motor.  We need a method to turn it on, then turn it off.  This will avoid the stage/unstage complications mentioned above.  We'll need two new plans to complement so this can be operated form a command list file (`usaxs.mac`) and in remote mode.

**Comment by prjemian** [comment]
@jilavsky We should make a decision about this branch.  Either modernize it or dump it.  Is the JPEG support (in-position sample photography) working for you?  Maybe just dump this PR then?

**Comment by jilavsky** [comment]
I believe jpg images are working. The only reason we wanted to have this "no waiting" done is to save time. It is less than 1 second per data collection, so it is not critical for now. It may get eventually more important, if we get to faster moves and faster data collections. Let's not bother about it for now.

**Comment by prjemian** [comment]
Obvious this will not be merged, closing.

**Pull request closed without merging**